components/hab/src/command/bldr/job/promote.rs

```diff
@@ -240,12 +243,25 @@ mod test {
                                                    "Properly formated timestamp".to_string(),
                                                project_name: "Test Project".to_string(),
                                                target:       "x86_64-linux".to_string(), };
+
+        #[cfg(windows)]
+        let tmp_file_path = {
+            let mut t = env::temp_dir();
```

Comment:
Why not use the [`tempfile`](https://docs.rs/tempfile/3.0.8/tempfile/) crate here? It would avoid the need to manually remove it later (which could get missed if `get_ident_list` fails)

Reply:
After looking at `test_get_ident_list_interactive` for a bit, I think we should remove this test. It's not actually testing the interactive-specific code (and this isn't really the right level to do such testing anyhow), so it's entirely redundant with `test_get_ident_list`, but adds unnecessary complexity.

Reply:
Cool. I'll remove it, but just to answer why I did not use TempFile: TempFile holds a handle to the file which gets removed on `drop` but I deenen the file handle to be closed bedore calling the file and `close` of TempFile will delete the file.

Reply:
Sounds good; I think removal is the best option. For my own understanding, could we have used `TempFile` (or probably `NamedTempFile::new_in` since it needed to be called `type.bat`) if we made the scope of `tmp_file_path` the entire function? Something like:
```rust
        #[cfg_attr(not(windows), allow(unused_variables))]
        let tmp_file_path: std::path::PathBuf;
        #[cfg(windows)]
        {
            // assign `tmp_file_path`
        }
```

Reply:
yeah that would have worked as long as we called `persist` to close it without deleting it but then we  would still needed to delete is when done.

Reply:
The original code is gone, so I can't confirm this, but wasn't `tmp_file_path` deleted before the end of that function? Unlike [`tempfile`](https://docs.rs/tempfile/3.0.8/tempfile/fn.tempfile.html), which
> will be automatically removed by the OS when the last handle to it is closed

[`NamedTempFile`](https://docs.rs/tempfile/3.0.8/tempfile/struct.NamedTempFile.html) stays around until it goes out of scope, so I think that would've fit the use case here. Just FYI for the future.